python/tvm/relay/frontend/pytorch.py

```diff
@@ -614,6 +615,51 @@ def _impl(inputs, input_types):
         return _op.tensor.sqrt(data)
     return _impl
 
+def _floor():
+    def _impl(inputs, input_types):
+        data = inputs[0]
+        return _op.floor(data)
+    return _impl
+
+def _to():
+    def _impl(inputs, input_types):
+        data = inputs[0]
+        # special handling for aten::to(data, 6, _, _, _) case
+        # 6 means dtype = float
+        # this happens when converting upsampling with scale factor
+        cast_func = {
+            6: float,
+        }
+        if inputs[1] in cast_func and not isinstance(data, _expr.Expr):
```

Comment:
This seems very specific, it looks like it only works for non expression inputs and casts to floats. Is there any reason not to add a `relay.cast` fallback for expression inputs and support other datatypes in the cast_func lookup?

Reply:
yeah, the problem is `aten::to(...)` is overloaded for different purpose (transfer to cpu/cuda or simple type cast). See my last commit https://github.com/apache/incubator-tvm/pull/4961/commits/f070d22fdb479c82160cf642e29453472e83c61e
Does this look better?

Reply:
Ah I didn't know it was overloaded for device copies. This does look much better though, thanks!